internal/pkg/options/validators.go

```diff
@@ -0,0 +1,33 @@
+package options
+
+import (
+	"errors"
+
+	"github.com/buzzfeed/sso/internal/pkg/sessions"
+)
+
+var (
+	ErrInvalidEmailAddress = errors.New("Invalid Email Address In Session State")
+	ErrValidationError     = errors.New("Error during validation")
+)
+
+type Validator interface {
+	Validate(*sessions.SessionState) error
+	Type() string
```

Comment:
This `Type() string` is a code smell. If we want to infer the type of something we should be using type checks rather than passing around a string.

Reply:
Removed this (check the previous comment for extra information)